models.py

```diff
@@ -70,8 +70,15 @@ def items(self, items):
                 self._unis.append(i)
             elif isinstance(i, Link):
                 self._links.append(i)
-            else:
+            elif isinstance(i, str):
```

Comment:
I think we should also check here if the switch is a valid switch on the controller list of switches. Something like this:

```suggestion
            elif isinstance(i, str) and i in self.controller.switches:
```

However, this will break a lot of tests